docs/architecture/server/Azure SQL read-only replica.md

```diff
@@ -0,0 +1,51 @@
+# Azure SQL read-only replica
+
+## Context
+
+Prior to April 2022, we had been noticing repeated errors that resemble the following:
+
+`Resource ID : 1. The request limit for the database is 300 and has been reached. See 'http://go.microsoft.com/fwlink/?LinkId=267637' for assistance.`
+
+To address the issue seen frequently during reconnects, we looked into our top query at the time and
+determined it to be the
+[User_ReadAccountRevisionDateById](https://github.com/bitwarden/server/blob/main/src/Sql/dbo/Stored%20Procedures/User_ReadAccountRevisionDateById.sql)
+stored procedure. Furthermore, the
+[readonlyconnection string setting](https://github.com/bitwarden/server/blob/5423e5d52fe2aac28db457107579aa91709c9bb7/src/Core/Settings/GlobalSettings.cs#L245-#L250)
+had been stubbed out already and needed to be utilized. Up until this point we had been utilizing
+the fallback mechanism, which meant it was using the default connection string to connect to the
+primary DB.
+
```

Comment:
🎨 I don't think this background is needed on the site and would be better served in the PR description to explain why it's being documented. You can then rephrase what's below this to state as fact that it's in use.